From 765fb52923dc5a02952724373ba367f413439a07 Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Mon, 17 Jun 2019 17:05:07 +0800
Subject: [PATCH 4/5] Fix CarService crash on booting

CarService crash due to CarBleTrustAgent try to access
the SharedPreferences failed. If the user doesn't unlock
the device, the SharedPreferences is a is an encrypted file.

The CarBleTrustAgent shouldn't access the SharedPreferences
until the user unlocks the device.

Change-Id: I33a17c240768d91e93e8142a8dbde65df35519b5
Tracked-On: https://jira.devtools.intel.com/browse/OAM-82632
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 .../android/car/trust/CarTrustAgentEnrollmentService.java  | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/service/src/com/android/car/trust/CarTrustAgentEnrollmentService.java b/service/src/com/android/car/trust/CarTrustAgentEnrollmentService.java
index 8f5b7a826..00d42ec5a 100644
--- a/service/src/com/android/car/trust/CarTrustAgentEnrollmentService.java
+++ b/service/src/com/android/car/trust/CarTrustAgentEnrollmentService.java
@@ -38,6 +38,8 @@ import android.content.Context;
 import android.content.SharedPreferences;
 import android.os.IBinder;
 import android.os.RemoteException;
+import android.os.UserHandle;
+import android.os.UserManager;
 import android.util.Log;

 import com.android.car.BLEStreamProtos.BLEOperationProto.OperationType;
@@ -313,6 +315,11 @@ public class CarTrustAgentEnrollmentService extends ICarTrustAgentEnrollment.Stu
     @NonNull
     @Override
     public List<TrustedDeviceInfo> getEnrolledDeviceInfosForUser(int uid) {
+        if (mContext.isCredentialProtectedStorage()
+                && !mContext.getSystemService(UserManager.class)
+                .isUserUnlockingOrUnlocked(UserHandle.myUserId())) {
+            return new ArrayList<TrustedDeviceInfo>(0);
+        }
         Set<String> enrolledDeviceInfos = mTrustedDeviceService.getSharedPrefs().getStringSet(
                 String.valueOf(uid), new HashSet<>());
         List<TrustedDeviceInfo> trustedDeviceInfos = new ArrayList<>(enrolledDeviceInfos.size());
--
2.21.0
